Makefile

```diff
@@ -67,9 +69,12 @@ build-natives-contained:
 	docker run --rm --entrypoint "/bin/sh" -v $(shell pwd):/binary $$image_id -c "cp /go/bin/mender-artifact* /binary" && \
 	docker image rm $$image_id
 
-install:
+goinstall:
 	@$(GO) install $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS)
 
+install: $(FINALEXE)
+	install -Dm0755 $(FINALEXE) $(DESTDIR)$(PREFIX)/bin/$(FINALEXE)
```

Comment:
We've had bad experience with this in the past, see [this commit](https://github.com/mendersoftware/mender/commit/b6e20f1f274e55fc09c9b0c09b5c8f9429ca2b05). `go install` caches the build artifacts, so I don't think this is very important for performance. Unless there's a strong reason I think it's better to keep this the same across the Mender repositories.

Reply:
I think there's a misunderstanding here. I changed the `install` target mainly to map it to the action of installing the final binary in a standard system-wide location (`$PREFIX/bin`, optionally prefixed with a `$DESTDIR` path useful when packaging), while before it was just called to build the final binary and put it under `$GOPATH/bin`.
Now, by looking at mendersoftware/mender@b6e20f1f2, I can see the concern regarding build tags change detection that can be mismanaged by only adding a target dependency on `FINALEXE`.
I can definitely look into it.

Reply:
Yes, the Go command would not obey `PREFIX` and `DESTDIR`.

I think this can be solved by simply replacing the dependency on `$(FINALEXE)` with `build` instead, and then making sure that `build` does not depend on it either (to avoid short circuiting the build).

Reply:
Removed the `$(FINALEXE)` dependency in favor of the phony `build` target dependency: this should enforce a fresh build on every `install` invocation.